### PR TITLE
Reformat code with fixed Indentation rule

### PR DESCRIPTION
**Pull request by arturbosch**
Needs #2709 to merge first.

@schalkms @BraisGabin I'm trying to fix all the formatting issue reports here.
With the last PR I noticed we used the wrong Indentation rule all the time.
With that fixed we also can now check for a consistent indentation on CI.
However we lose a lot of history. Should we still merge this?

 I've activated the rule in detekt.yml and format.yml

### CODE REVIEW

**Comment by schalkms** [comment]
We need to be careful. In the last days I noticed problems that occurred due to a reformatting commit. (#2694)
I don't think we should reformat everything again. I'm fine with using the old `Indentation` rule.

File: buildSrc/src/main/kotlin/packaging.gradle.kts [hunk 1]
```diff
@@ -33,29 +33,37 @@ subprojects {
...
+                                user = mavenCentralUser
+                                password = mavenCentralPassword
+                                close = "1"
+                            }
+                        )
+                    }
+                )
+            }
+        )
     }
```
Comment (BraisGabin) [inline buildSrc/src/main/kotlin/packaging.gradle.kts:67]: I don't like this kind of things. 6 lines that say nothing. 

```kotlin
                })
            })
        })
```

☝️ This is more easy to read.

**Comment by arturbosch** [comment]
> We need to be careful. In the last days I noticed problems that occurred due to a reformatting commit. (#2694)

This time no resources got formated.

> I don't think we should reformat everything again. I'm fine with using the old `Indentation` rule.

We had no Indentation rule :). That's why we still often have continuous indentation of 8 spaces :).

I'm okay with not activating these three rules (Indentation, NoEmptyFirstLineInMethodBlock and AnnotationOnSeparateLine). But the PR proves that detekt now reports the same issues as KtLint in the detekt repo itself.

**Comment by BraisGabin** [comment]
And now that I think about it... I should probably open an issue (or comment if it exists already) about this indention... I tried out in some projects and I disabled it all the times because of this. And if it's in experimental they should be looking for feedback...

**Comment by BraisGabin** [comment]
@Mkohm I think that the previous comment should be removed or updated. Otherwise this is too noisy.

File: detekt-test/src/main/kotlin/io/gitlab/arturbosch/detekt/test/FindingsAssertions.kt [hunk 2]
```diff
@@ -14,25 +14,25 @@ fun assertThat(finding: Finding) = FindingAssert(finding)
 fun List<Finding>.assert() = FindingsAssert(this)
 
 class FindingsAssert(actual: List<Finding>) :
-        AbstractListAssert<FindingsAssert, List<Finding>,
-                Finding, FindingAssert>(actual, FindingsAssert::class.java) {
+    AbstractListAssert<FindingsAssert, List<Finding>,
+        Finding, FindingAssert>(actual, FindingsAssert::class.java) {
 
     override fun newAbstractIterableAssert(iterable: MutableIterable<Finding>): FindingsAssert {
```
[the remark this answers is not in the document]
Comment (Mkohm) [reply detekt-test/src/main/kotlin/io/gitlab/arturbosch/detekt/test/FindingsAssertions.kt:20]: @BraisGabin , you mean this one?

**Comment by Mkohm** [comment]
@BraisGabin , @arturbosch - This is really helpful. Actually seeing the tool run on an actual project. This PR especially (reformatting a lot of files) revealed some problems that did go under the radar when developing. Looking into the execution of the last commit, detekt-hint was actually rate-limited because of the amount of comments that would have been added to this pull request.

image

I think i have enough feedback for another iteration solving these problems.

**Pull request closed without merging**